tests/junit.py

```diff
@@ -1,264 +1,128 @@
+    return parser
+
+
+# Necessary functions for running tests
+class CeedSuiteSpec(SuiteSpec):
+    def get_source_path(self, test: str) -> Path:
+        prefix, rest = test.split('-', 1)
+        if prefix == 'petsc':
+            return (Path('examples') / 'petsc' / rest).with_suffix('.c')
+        elif prefix == 'mfem':
+            return (Path('examples') / 'mfem' / rest).with_suffix('.cpp')
+        elif prefix == 'nek':
+            return (Path('examples') / 'nek' / 'bps' / rest).with_suffix('.usr')
+        elif prefix == 'fluids':
+            return (Path('examples') / 'fluids' / rest).with_suffix('.c')
+        elif prefix == 'solids':
+            return (Path('examples') / 'solids' / rest).with_suffix('.c')
+        elif test.startswith('ex'):
+            return (Path('examples') / 'ceed' / test).with_suffix('.c')
+        elif test.endswith('-f'):
+            return (Path('tests') / test).with_suffix('.f90')
+        else:
+            return (Path('tests') / test).with_suffix('.c')
+
+    # get path to executable
+    def get_run_path(self, test: str) -> Path:
+        return Path('build') / test
+
+    def get_output_path(self, test: str, output_file: str) -> Path:
+        return Path('tests') / 'output' / output_file
+
+    def check_pre_skip(self, test: str, spec: TestSpec, resource: str, nproc: int) -> Optional[str]:
+        return 'Pre-Check Skipped' if any((
+            test.startswith('t4') and contains_any(resource, ['occa']),
+            test.startswith('t5') and contains_any(resource, ['occa']),
+            test.startswith('ex') and contains_any(resource, ['occa']),
+            test.startswith('mfem') and contains_any(resource, ['occa']),
+            test.startswith('nek') and contains_any(resource, ['occa']),
+            test.startswith('petsc-') and contains_any(resource, ['occa']),
+            test.startswith('fluids-') and contains_any(resource, ['occa']),
+            test.startswith('solids-') and contains_any(resource, ['occa']),
+            test.startswith('t318') and contains_any(resource, ['/gpu/cuda/ref']),
+            test.startswith('t506') and contains_any(resource, ['/gpu/cuda/shared']),
+        )) else None
+
+    def check_post_skip(self, test: str, spec: TestSpec, resource: str, stderr: str) -> Optional[str]:
+        if 'OCCA backend failed to use' in stderr:
+            return f'occa mode not supported {test} {resource}'
+        elif 'Backend does not implement' in stderr:
+            return f'not implemented {test} {resource}'
+        elif 'Can only provide HOST memory for this backend' in stderr:
+            return f'device memory not supported {test} {resource}'
+        elif 'Test not implemented in single precision' in stderr:
+            return f'not implemented {test} {resource}'
+        elif 'No SYCL devices of the requested type are available' in stderr:
+            return f'sycl device type not available {test} {resource}'
+        return None
```

Comment:
We can take the opportunity no make this more human readable

Reply:
updated them with more readable output, as well as for the pre-skip reasons